kong/tools/queue.lua

```diff
@@ -0,0 +1,426 @@
+--[[
+TODO:
+
+queue objects cannot be held on to because they may be garbage collected.  safer API?
+update documentation
+metrics
+when at capacity, remove currently sent *batch* from the front
+use in/out pointer instead of array manipulation
+]]
+
+
+
+-- Queue with background retryable processing
+--
+-- This is a queue of "entries". Entries can be any Lua value.
+--
+-- The queue has two mandatory parameters:  `name` is the name of the queue,
+-- used to identify it in log messages, and `handler`, which is the
+-- function which consumes entries.  Entries are dequeued in batches.  The
```

Comment:
Good documentation. I recommend adding a bit more context to `name` - are queues unique based on names or are they simply a helper attribute.